### PR TITLE
Feature/misc improvements

### PR DESCRIPTION
**Pull request by tonihele**
- Imps drop the excess gold when digging at their feet if there is no place to store it currently. Prevents imps from coming millionaires
- Creature fleeing is a bit nicer now, always annoyed me the they could not decide what to do, now at least they are a bit more decisive and stay fleeing a minimum time

### CODE REVIEW

**Comment by tonihele** [comment]
I don't actually have now the original game at hand (Linux...) so I can't check whether the gold clumped up to bigger piles (the max pile size) on the same tiles or just like this. A lot of small clumps. This has an actual effect on the game as the imp tasks are not grouped. The smaller the piles, the less effective your imps are collecting it...

File: src/toniarts/openkeeper/game/controller/creature/CreatureController.java [hunk 1]
```diff
@@ -858,7 +858,21 @@ public void resetReEvaluationTimer() {
     @Override
     public void addGold(int amount) {
         Gold gold = entityData.getComponent(entityId, Gold.class);
-        entityData.setComponent(entityId, new Gold(gold.gold + amount, gold.maxGold));
+
+        // Drop excess gold, we can only carry so much
+        boolean hasMaxGold = gold.maxGold > 0;
+        int maxGoldCanAdd = hasMaxGold ? gold.maxGold - gold.gold : Integer.MAX_VALUE;
```
Comment (Trass3r) [inline src/toniarts/openkeeper/game/controller/creature/CreatureController.java:864]: Why isn't MAX_VALUE used to designate no limit in the first place?

Comment (tonihele) [reply src/toniarts/openkeeper/game/controller/creature/CreatureController.java:864]: That is a question I ask myself every night I go to sleep... :D Yeah, I'll change it. The logic is still a bit iffy here anyway, I didn't even check that do normal creatures have a max value they can carry? Essentially dropping all bribes and pay they are given if they didn't have a chance to blow it all in the rigged casino...

Comment (tonihele) [reply src/toniarts/openkeeper/game/controller/creature/CreatureController.java]: Ok, I checked. Creatures always have some max gold set, everybody. 0 is not used to indicate max gold. That was my half assed assumption. So removed all assumptions from here.

Also normal creatures do have max gold. I guess it makes sense in the real world. But I don't know how the normal creatures deal with their possessions...

**Pull request merged**